src/content/configuration/entry-context.md

```diff
@@ -33,7 +34,7 @@ By default the current directory is used, but it's recommended to pass a value i
 
 ## `entry`
 
-`string` `[string]` `object = { <key> string | [string] }` `(function() => string | [string] | object = { <key> string | [string] })`
+`string` `[string]` `object = { <key> string | [string] | object = { import string | [string], dependOn string | [string], filename string }}` `(function() => string | [string] | object = { <key> string | [string] } | object = { import string | [string], dependOn string | [string], filename string })`
```

Comment:
@EugeneHlushko This might grow out of control, we should think in a different way to handle this.

Reply:
Why we don’t use ts types?

Reply:
Originally we wanted to keep webpack docs typings short which were somewhat similar to what you see now, what we've done in the past is making them consistent and documenting them in contribute section. But later as we started being precise on types (previously this wasnt the case for the whole documentation), types started to look "longer" and "even longer" :D

There was a suggestion to convert it to TS but its still not clear if TS would look shorter, i think TS or this both has disadvantages, you dont say that entry level developer is going to understand typescript? This is also not the best approach in terms of accessibility of the docs. In the end, to use the tool, you dont need to understand its internals

Reply:
@EugeneHlushko I think types should be provided and currently how webpack docs displays type seems perfect. I wouldn't suggest to convert it to TS cause I **think** it will be much longer cause of external types and interfaces

Current type placements are perfect as well cause developers usually don't want to navigate or open a new tab just to see what kind of information it takes. 

Perhaps ! can we create a separate section of the page for types to link long types like this one there? (*again navigating point will oppose this*)

What I think is this is good. cause longer types are handy as long as it is distance itself from the main content.

What do you think ?

Reply:
I have an idea:
Extract, generate and export webpack source TS types to an static page, then link it from the docs.

Another one:
Extract types information into a JSON format, then embed it into the docs through React.

Everything automated.

Reply:
> Perhaps ! can we create a separate section of the page for types to link long types like this one there? (again navigating point will oppose this)

like this ?

> Perhaps ! can we create a separate section of the page for types to link long types like this one there? (again navigating point will oppose this)

Reply:
@montogeek webpack 's source has jsdocs types comment 

using [`this`](https://jsdoc.app/about-getting-started.html#generating-a-website)

we can add something like `type.webpack.js.org` ?